### PR TITLE
feat(@size-limit/esbuild-why): support multiple sections for `--why`

### LINKED ISSUE #311: multiple checks with `--why` only kept last `report.html`
e.g.:

```
// package.json
{
  "size-limit": [
    {
      "path": "./cjs/index.js",
      "limit": "15 kb"
    },
    {
      "path": "./esm/index.js",
      "limit": "15 kb"
    },
    {
      "path": "./tslib/index.js",
      "limit": "15 kb"
    }
  ]
}
```

Running `size-limit --why` (using `esbuild` and `esbuild-why`) will create a `report.html` containing the result of the last one (`tslib`) as the previous ones are overwritten.

I can change it to add `report-{x}.html`, but a better way is giving each check a name, and name the output as `report-{name}.html`?

Currently the `check` contains a `name`, but it is the entry file name:

```
{
  path: './tslib/index.js',
  limit: '15 kb',
  files: [ 'D:\\code\\unional\\type-plus\\tslib\\index.js' ],
  name: './tslib/index.js',
  sizeLimit: 15000,
  ...
}
```

### PR DESCRIPTION
**Pull request by unional**
Fixed #311

### CODE REVIEW

File: packages/esbuild-why/index.js [hunk 1]
```diff
@@ -1,14 +1,16 @@
-let { visualizer } = require('esbuild-visualizer')
-let { join } = require('path')
-let { writeFileSync } = require('fs')
+const { visualizer } = require('esbuild-visualizer')
```
Comment (ai) [inline packages/esbuild-why/index.js:1]: I use `let` :)

File: packages/esbuild-why/report.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const isValidFilename = require('valid-filename')
```
Comment (ai) [inline packages/esbuild-why/report.js:1]: The check is too simple to use another dependency. Let’s just copy regexp

https://github.com/sindresorhus/filename-reserved-regex/blob/main/index.js

File: README.md [hunk 3]
```diff
@@ -573,6 +573,39 @@ inserts `style-loader` runtime (≈2 kB) into the bundle.
 [Statoscope docs]: https://github.com/statoscope/statoscope/tree/master/packages/webpack-plugin#optionsreports-report
 [pattern]: https://github.com/sindresorhus/globby#globbing-patterns
 
+## Analyze with `--why`
+
+You can run `size-limit --why` to analyze the bundle.
+
+You will need to install `@size-limit/esbuild-why` or `@size-limit/webpack-why` depends on which bundler you are using (default is `esbuild`).
```
Comment (ai) [inline README.md:580]: We need to keep 80 columns line

File: packages/esbuild-why/index.js (hunk 1 above)
Comment (unional) [reply packages/esbuild-why/index.js:1]: Oh, also at the top-level?

I looked up the `eslint-plugin-prefer-let` it says you can use `const` at top-level. 😛 

I can change that.

Comment (ai) [reply packages/esbuild-why/index.js:1]: I use `const` only for cases like `const MAX = 100`

File: packages/esbuild-why/report.js (hunk 2 above)
Comment (unional) [reply packages/esbuild-why/report.js]: I pulled in the code directly. Added attribution

**Comment by ai** [comment]
Thanks. Released in 8.2.4.

**Pull request merged**